src/core/unif.ml

```diff
@@ -184,19 +229,32 @@ let rec solve : problem -> constr list = fun p ->
 
 and solve_aux : ctxt -> term -> term -> problem -> constr list =
   fun ctx t1 t2 p ->
+  let initial = (ctx,t1,t2)::p.to_solve in
   let t1 = Eval.whnf ctx t1 in
   let t2 = Eval.whnf ctx t2 in
   let (h1, ts1) = Basics.get_args t1 in
   let (h2, ts2) = Basics.get_args t2 in
   if !log_enabled then log_unif "solve %a" pp_constr (ctx, t1, t2);
 
   let add_to_unsolved () =
-    if Eval.eq_modulo ctx t1 t2 then solve p else
-    match try_rules ctx t1 t2 with
-    | None     -> solve {p with unsolved = (ctx,t1,t2) :: p.unsolved}
-    | Some([]) -> assert false
-    (* Unification rules generate non empty list of unification constraints *)
-    | Some(cs) -> solve {p with to_solve = cs @ p.to_solve}
+    if Eval.eq_modulo ctx t1 t2 then
+      solve p
+    else
+      match try_rules ctx t1 t2 with
+      | None     -> solve {p with unsolved = (ctx,t1,t2) :: p.unsolved}
+      | Some([]) -> assert false
+      (* Unification rules generate non empty list of unification
+         constraints *)
+      | Some(cs) ->
+        let lost_constrs = []
+(*
+          try (* TODO to get constraints on the type of implicit argument *)
```

Comment:
what are those lines (let and comment) for?

Reply:
this part has been removed